modules/data-loading/pages/data-streaming-connector/azure-blob.adoc

```diff
@@ -0,0 +1,152 @@
+Connector class is specified by the `connector.class` key.
+For connecting to ABS, its value is
+`connector.class=com.tigergraph.kafka.connect.filesystem.azure.AzureBlobSourceConnector`.
+
+==== Storage account credentials
+
+You have two options when providing credentials:
+
+[tabs]
+====
+Shared key authentication::
++
+--
+Shared key authentication requires you to provide your storage account name and account key.
+
+You can find both the account name and account key on the Access Key tab of your storage account:
+
+.Azure storage account page Access Keys tab
+image::azure-storage-account.png[Azure storage account page Access Keys tab]
+
+To specify the account name and key, use the following configuration.
+Replace `<account_name>` with your account name and replace `<account_key>` with your account key:
+
+[.wrap,text]
+----
+file.reader.settings.fs.azure.account.key.<account_name>.dfs.core.windows.net="<account_key>"
+----
+
+--
+Service principal authentication::
++
+--
+To use service principal authentication, you must first register your TigerGraph instance as an application provide it access to your storage account.
+
+Once the application is registered, you'll need the following credentials:
+
+* `CLIENT_ID`
+* `CLIENT_SECRET`
+* `TENANT_ID`
+
+Provide the credentials using the following configs:
+
+[.wrap]
+----
+file.reader.settings.fs.azure.account.oauth2.client.id="4a3d45fa-8974-48e6-af97-2614b1a24be8" <1>
+file.reader.settings.fs.azure.account.oauth2.client.secret="Ydt7Q~TRfxvuMxEDgn4mwGKg9TLhaKywkGjwn" <2>
+file.reader.settings.fs.azure.account.oauth2.client.endpoint="https://login.microsoftonline.com/718f143b-b1ef-468b-a3a8-c95651fdd511/oauth2/token"<3>
+----
+<1> Client ID.
+<2> Client secret.
+<3> Tenant ID.
+--
+====
+
+==== Other credentials
+The connector uses Hadoop to connect to the Azure File system.
+The configurations below are required along with our recommended values:
+
+----
+file.reader.settings.fs.abfs.impl="org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem"
+file.reader.settings.fs.abfss.impl="org.apache.hadoop.fs.azurebfs.SecureAzureBlobFileSystem"
+file.reader.settings.fs.AbstractFileSystem.abfs.impl=""org.apache.hadoop.fs.azurebfs.Abfs
+file.reader.settings.fs.AbstractFileSystem.abfss.impl="org.apache.hadoop.fs.azurebfs.Abfss"
+----
+
+include::partial$parsing-rules.adoc[]
+
+include::partial$map-source-to-connector.adoc[]
+
+URIs for files must be configured in the following way:
+
+`<protocol: abfss | abfs>://<container name>@<accountname>.dfs.core.windows.net/<path to the file>`
+
+For example:
+
+`abfss://person@yandblobstorage.dfs.core.windows.net/persondata.csv`
+
+==== Example
+
+The following is an example configuration file:
+
+----
+connector.class=com.tigergraph.kafka.connect.filesystem.azure.AzureBlobSourceConnector
+file.reader.settings.fs.defaultFS="abfss://person@yandblobstorage.dfs.core.windows.net/"
```

Comment:
We should give it a fake name for file system and account name in the url. See https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction-abfs-uri for details